libsolidity/formal/SymbolicTypes.cpp

```diff
@@ -169,3 +169,32 @@ smt::Expression dev::solidity::maxValue(IntegerType const& _type)
 {
 	return smt::Expression(_type.maxValue());
 }
+
+void dev::solidity::setZeroValue(SymbolicVariable const& _variable, smt::SolverInterface& _interface)
+{
+	setZeroValue(_variable.currentValue(), _variable.type(), _interface);
+}
+
+void dev::solidity::setZeroValue(smt::Expression _expr, TypePointer const& _type, smt::SolverInterface& _interface)
+{
+	if (isInteger(_type->category()))
+		_interface.addAssertion(_expr == 0);
+	else if (isBool(_type->category()))
+		_interface.addAssertion(_expr == smt::Expression(false));
+}
```

Comment:
Should this assert in other cases?

Reply:
No, the other cases just lack information and might give false positives